src/black/__init__.py

```diff
@@ -469,7 +469,13 @@ def main(  # noqa: C901
         out(main.get_usage(ctx) + "\n\nOne of 'SRC' or 'code' is required.")
         ctx.exit(1)
 
-    root, method = find_project_root(src) if code is None else (None, None)
+    if stdin_filename is not None:
+        src_with_stdin_filename = tuple(stdin_filename if s == "-" else s for s in src)
+    else:
+        src_with_stdin_filename = src
+    root, method = (
+        find_project_root(src_with_stdin_filename) if code is None else (None, None)
+    )
```

Comment:
```suggestion
    if stdin_filename is not None:
        src = tuple(stdin_filename if s == "-" else s for s in src)

    root, method = find_project_root(src) if code is None else (None, None)
```

Perhaps we could make the above work below as well.

Also, I was unaware how passing multiple stdin placeholders works, but for `- - -` etc stdin is only read once (or does closing it prevent further reads, guess so), and with other files `-` is read as a filename. So it is safe to assume that there will only be one to convert. Or maybe we'll have to even guard against weird use, not sure what would be best.

Yeah, tests would be nice 👍